DXMainClient/Domain/Multiplayer/CnCNet/MapSharer.cs

```diff
@@ -365,18 +365,27 @@ public static string GetMapFileName(string sha1, string mapName)
 
         private static string DownloadMain(string sha1, string myGame, string mapName, out bool success)
         {
-            string customMapsDirectory = SafePath.CombineDirectoryPath(ProgramConstants.GamePath, "Maps", "Custom");
+            string customMapsPath = SafePath.CombineDirectoryPath(ProgramConstants.GamePath, "Maps", "Custom");
+            string tempDownloadPath = SafePath.CombineDirectoryPath(ProgramConstants.GamePath, "Maps", "temp");
```

Comment:
Use a temporary download directory so that the filewatcher doesn't prematurely load the `.map` file before it is fully extracted and renamed.